File: codengine-cli/test/fixtures/cs-project/Tasks.cs
// Plain public static C# methods — no adaptation for codengine. The analyzer reads
// the parameter names and the runner binds the input to them by name (reflection).
namespace Greeting;

public static class Tasks
{
    public static Dictionary<string, object?> greet(string name)
        => new() { ["message"] = $"Hello, {name}!" };

    public static Dictionary<string, object?> output(string message)
        => new() { ["message"] = message };
}

File: codengine-cli/test/fixtures/dart-project/tasks.dart
// Plain top-level Dart functions — no adaptation for codengine. `output` is `async`
// (returns a Future); codengine accepts sync and async task functions alike.
Map<String, dynamic> greet({required String name}) => {'message': 'Hello, $name!'};

Future<Map<String, dynamic>> output(Map<String, dynamic> input) async => input;
